src/bindings/html/langs.js

```diff
@@ -28,33 +28,50 @@ export function onlanguagechage(
 export function onadditionallanguageschange(
   appVersion, defaultLang, availableLangs, additionalLangs, requestedLangs) {
 
-  return this.languages = this.languages.then(
+  this.languages.then(
     prevLangs => changeLanguage.call(
       this, appVersion, defaultLang, availableLangs, additionalLangs,
       prevLangs, requestedLangs || navigator.languages));
 }
 
-
-export function changeLanguage(
-  appVersion, defaultLang, availableLangs, additionalLangs, prevLangs,
-  requestedLangs) {
+export function getLanguageChain(appVersion, defaultLang, availableLangs,
+    additionalLangs, prevLangs, requestedLangs) {
 
   let allAvailableLangs = Object.keys(availableLangs).concat(
     additionalLangs || []).concat(Object.keys(qps));
   let newLangs = prioritizeLocales(
     defaultLang, allAvailableLangs, requestedLangs);
 
-  let langs = newLangs.map(code => ({
+  return newLangs;
+}
+
+export function getLangStruct(langs, appVersion, availableLangs,
+  additionalLangs) {
+
+  return langs.map(code => ({
     code: code,
     src: getLangSource(appVersion, availableLangs, additionalLangs, code),
     dir: getDirection(code)
   }));
+}
 
-  if (!arrEqual(prevLangs, newLangs)) {
-    initViews.call(this, langs);
+export function changeLanguage(
+  appVersion, defaultLang, availableLangs, additionalLangs, prevLangs,
+  requestedLangs) {
+
+  let newLangs = getLanguageChain(appVersion, defaultLang, availableLangs,
+    additionalLangs, prevLangs, requestedLangs);
+
+
+  if (prevLangs[0] === newLangs[0]) {
```

Comment:
But if we end up doing that, there should also be a code path for changing user preferences that uses `arrEqual`.